docs/LOCALIZATION.md

```diff
@@ -0,0 +1,251 @@
+
+## Supported languages
+
+Overview of the languages supported by Enterprise Aurora and their matching translation file names.
+
+| Language | Filename |
+| ------------- | ------------- |
+| Czech  | `csCZ.json` |
+| German  | `deDE.json` |
+| English  | `enUS.json` |
+| Spanish  | `esES.json` |
+| Finnish  | `fiFI.json` |
+| French  | `frFR.json` |
+| Italian  | `jaJP.json` |
+| Korean  | `koKR.json` |
+| Dutch  | `nlNL.json` |
+| Polish  | `plPL.json` |
+| Portuguese  | `ptBR.json` |
+| Russian  | `ruRU.json` |
+| Simplified Chinese  | `zhCN.json` |
+| Traditional Chinese  | `zhTW.json` |
+
+## Localization of `components-definition.json`
+
+### Component with a hardcoded UI string
+
+The label property simply contains a string which will be shown directly.
+
+```javascript
+{
+    "name": "body",
+    "label": "Body",
+    "icon": "icons/components/body.svg",
+    // ...
+},
+```
+
+### Component with a localized UI string
+
+The label property contains an object containing a key property:
+
+```javascript
+{
+    "name": "body",
+    "label": { "key": "COMPONENT_BODY_LABEL" },
+    "icon": "icons/components/body.svg",
+    // ...
+},
+```
+
+The key points to an translation in for example `enUS.json`:
+
+```javascript
+{
+    // ...
+    "COMPONENT_BODY_LABEL": "Body",
+    // ...
+}
+```
+
+Keys for which a translation has not been given will be replaced by the key itself.
+
+### Translations with variable parts
+
+Translations can contain interpolations that will be substituted by the values provided. The label property will have an object that besides a key property has a values property containing the variable names and replacement values:
+
+```javascript
+{
+	"name": "style",
+	// ...
+	"control": {
+		"type": "select",
+		"options": [
+			{
+				"caption": { "key": "PROPERTY_STYLE_DEFAULT_OPTION" }
+			},
+			{
+				"caption": { "key": "PROPERTY_VARIANT_STYLE_OPTION", "values": { "0" : "1"} },
+				"value": "_option1"
+			},
+			{
+				"caption": { "key": "PROPERTY_VARIANT_STYLE_OPTION", "values": { "0" : "2"} },
+				"value": "_option2"
+			},
+			// ...
+		]
+	},
+	"dataType": "styles"
+},
+```
+
+The translation in for example `enUS.json` contains the interpolations:
+
+```javascript
+{
+    // ...
+    "PROPERTY_VARIANT_STYLE_OPTION": "Option {{0}}",
+    // ...
+}
+```
+
+Interpolations for which a value has not been given will be replaced by the content of the interpolation itself.
+
+### Overview of all localizable fields
+
+In the next sections all fields that can be localized are enumerated.
+
+#### Component UI name
+
+Used in the component chooser and as title in the properties side bar.
+
+```javascript
+{
+    "name": "title",
+    "label": { "key": "COMPONENT_TITLE_LABEL" },
+    "icon": "icons/components/title.svg",
+    // ...
+},
+```
+
+#### Component properties header
```

Comment:
It could actually apply to any property, not just the header type. componentProperties is merged into the properties list of each component. You can define any property in this list.

Reply:
Fair, I'll use the header as example instead. I do think we need to describe better in `COMPONENTS.md` how it works.

Reply:
I chose to remove the header one entirely